pkg/analysis_server/lib/src/plugin/plugin_watcher.dart

```diff
@@ -132,7 +132,7 @@ class PluginWatcher implements DriverWatcher {
   String _getSdkPath(AnalysisDriver driver) {
     AbsolutePathContext context = resourceProvider.absolutePathContext;
     String sdkRoot = driver.sourceFactory.forUri('dart:core').fullName;
-    while (sdkRoot.isNotEmpty && context.basename(sdkRoot) != 'lib') {
+    while (sdkRoot.isNotEmpty && sdkRoot != '/' && context.basename(sdkRoot) != 'lib') {
```

Comment:
Unfortunately, this won't work on Windows. Perhaps revert the condition and replace the body of the `while` with
```
String parent = context.dirname(sdkRoot);
if (parent == sdkRoot) {
   break;
}
sdkRoot = parent;
```